src/Containers/Microsoft.NET.Build.Containers/ImageConfig.cs

```diff
@@ -37,6 +40,11 @@ internal sealed class ImageConfig
     public ReadOnlyDictionary<string, string> EnvironmentVariables => _environmentVariables.AsReadOnly();
     public HashSet<Port> Ports => _exposedPorts;
 
+    /// <summary>
+    /// Represents the platform information of this image, including architecture, os, variant, os features and os version.
+    /// </summary>
+    public PlatformInformation PlatformInformation => new(_architecture, _os, _variant, _osFeatures ?? [], _osVersion);
```

Comment:
With all of the new data, we can synthesize our PlatformInformation structure for compat-checking purposes.